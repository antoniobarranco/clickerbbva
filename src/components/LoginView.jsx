import React from 'react';
import { useNavigate } from 'react-router-dom';

import useForm from '../hooks/useForm';
import BreadCrumbs from './ui/BreadCrumbs';

function LoginView() {
  const [formValues, handleInputChange] = useForm({ userName: '' });

  const { userName } = formValues;

  const navigate = useNavigate();

  const handlePlay = (e) => {
    e.preventDefault();
    navigate(`/game?player=${userName.toLowerCase()}`);
  };

  const rankingHandle = () => {
    navigate('/ranking', { replace: true });
  };

  return (

    <>
      <BreadCrumbs Text="Please, enter your name..." />

      <div className="d-grid gap-2 col-6 my-5 mx-auto">

        <form onSubmit={handlePlay}>

          <div className="input-group mb-3">
            <input type="text" name="userName" required maxLength="25" minLength="3" pattern="[a-zA-Z0-9-]+" value={userName} onChange={handleInputChange} className="form-control" placeholder="Put your name" aria-label="User name to play" aria-describedby="button-submit" />
            <button className="btn btn-primary" type="submit" id="button-submit"><i className="bi bi-joystick"></i> Let&apos;s Play</button>
          </div>

        </form>

        <div className="alert alert-primary" role="alert">You can choose a name composed by letters and numbers. Minimum size is 3 and maximum 25. Remember that if you already played your progress was saved.</div>

        <button type="button" className="btn btn-primary btn-lg" onClick={rankingHandle}><i className="bi bi-bar-chart-line-fill"></i> Rankings</button>

      </div>
    </>

  );
}

export default LoginView;
