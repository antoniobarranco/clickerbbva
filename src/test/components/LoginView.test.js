import renderer from 'react-test-renderer';
import { render, fireEvent, screen } from "@testing-library/react";
import LoginView from "../../components/LoginView";

describe('Tests on LoginView component', () => {

    test('Test LoginView with Snapshot test for UI changes', () => {

    const tree = renderer.create(<LoginView />).toJSON();

    expect(tree).toMatchSnapshot();

    })

    test('Test LoginView inputs lets write on it', () => {

      const name = 'Username';

      render(<LoginView />);

      const input = screen.getByPlaceholderText(/Put your name/i);

      fireEvent.change(input, {target: {value: name}})
      expect(input.value).toBe(name);

    })

})