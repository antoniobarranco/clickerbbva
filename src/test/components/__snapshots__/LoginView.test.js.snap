// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on LoginView component Test LoginView with Snapshot test for UI changes 1`] = `
Array [
  <nav
    className="navbar navbar-dark bg-primary bg-opacity-75"
  >
    <div
      className="container-fluid"
    >
      <span
        className="navbar-brand mb-0 h1"
        data-testid="BreadCrumbsText"
      >
        Please, enter your name...
      </span>
    </div>
  </nav>,
  <div
    className="d-grid gap-2 col-6  my-5 mx-auto"
  >
    <form
      data-testid="FormLogin"
      onSubmit={[Function]}
    >
      <div
        className="input-group mb-3"
      >
        <input
          aria-describedby="button-submit"
          aria-label="User name to play"
          className="form-control"
          max="25"
          min="3"
          name="userName"
          onChange={[Function]}
          pattern="[a-zA-Z0-9-]+"
          placeholder="Put your name"
          required={true}
          type="text"
          value=""
        />
        <button
          className="btn btn-primary"
          data-testid="ButtonLogin"
          id="button-submit"
          type="submit"
        >
          <i
            className="bi bi-joystick"
          />
           Let's Play
        </button>
      </div>
    </form>
    <div
      className="alert alert-primary"
      role="alert"
    >
      You can choose a name composed by letters and numbers. Minimum size is 3 and maximum 25. Remember that if you already played your progress was saved.
    </div>
    <button
      className="btn btn-primary btn-lg"
      data-testid="ButtonRankings"
      onClick={[Function]}
      type="button"
    >
      <i
        className="bi bi-bar-chart-line-fill"
      />
       Rankings
    </button>
  </div>,
]
`;
